app.js

```diff
@@ -17,11 +17,15 @@ app.get('/api/articles', getAllArticles)
 
 app.get('/api/articles/:article_id/comments', getCommentsByArticleId)
 
+app.post('/api/articles/:article_id/comments', addComment);
+
 app.all('*', (req, res) => {
-    res.status(404).send({ msg: 'Route not found' });
+    res.status(404).send({ msg: 'Route not found' })
   })
 
-
-
+  app.use((err, req, res, next) => {
+    res.status(err.status || 500).send({ msg: err.msg || 'Internal Server Error' })
```

Comment:
Interesting approach here! What would be the benefits of extracting the logic out to separate error handlers?